### ISSUE TITLE
Deprecate `asyncio.iscoroutinefunction`

### ISSUE THREAD

**Issue by kumaraditya303** [python/cpython#122858]
`asyncio.iscoroutinefunction` should be deprecated in favor of `inspect.iscoroutinefunction`, we don't need two different functions to do the same thing. I propose to deprecate `asyncio.iscoroutinefunction` because IMO these introspection functions don't belong to `asyncio` but more to `inspect` module. It will be deprecated in Python 3.14 and scheduled for removal in Python 3.16.

### Linked PRs
* gh-122875

**Comment by Wulian233** [comment]
See https://github.com/python/cpython/pull/94923

**Comment by Eclips4** [comment]
Looks like an duplicate of #94912

**Comment by kumaraditya303** [comment]
> Looks like an duplicate of #94912

No it isn't, in this issue we are just deprecating it not making an alias.

**Comment by illia-v** [comment]
If you get the deprecation error and want to support Python < 3.12, you may need to still use `asyncio.iscoroutinefunction` conditionally for backward compatibility because before https://github.com/python/cpython/pull/94050 merged into 3.12 (and only some 3.10 and 3.11 releases) the functions hadn't been equivalent

```python
if sys.version_info >= (3, 12):
    iscoroutinefunction = inspect.iscoroutinefunction
else:
    iscoroutinefunction = asyncio.iscoroutinefunction
```
